include/jni/arraylike.hpp

```diff
@@ -30,8 +30,11 @@ namespace jni
     template < class E, std::size_t n >    E const * ArraylikeData(const std::array<E,n>& a)              { return a.data(); }
     template < class E, class A >          E       * ArraylikeData(      std::vector<E,A>& a)             { return a.data(); }
     template < class E, class A >          E const * ArraylikeData(const std::vector<E,A>& a)             { return a.data(); }
-    template < class C, class T, class A > C       * ArraylikeData(      std::basic_string<C,T,A>& a)     { return &a[0]; }
-    template < class C, class T, class A > C const * ArraylikeData(const std::basic_string<C,T,A>& a)     { return &a[0]; }
+    template < class C, class T, class A > C       * ArraylikeData(      std::basic_string<C,T,A>& a)     { return a.data(); }
```

Comment:
The reason to use `&a[0]` as opposed to `a.data()` is that [the non-`const` overload for `data()`](http://en.cppreference.com/w/cpp/string/basic_string/data) is provided only in C++17, and this library maintains compatibility with C++14. Could you revert this change please?

Reply:
a[0] will not work with empty strings though :(  Wouldn't it be better to condition it on C++17?

Reply:
It does work (since C++11): it's defined to return a reference to the character with value CharT() (the null character).